### PR TITLE
ENH: Transition project configuration to `pyproject.toml`

### PR DESCRIPTION
**Pull request by jhlegarreta**
Transition project configuration to `pyproject.toml`.

Used the `setup-py-to-pyproject-toml.py` script in the `ITKModuleTemplate` repository:
https://github.com/InsightSoftwareConsortium/ITKModuleTemplate

as a starting base, then made manual changes as necessary.

Require `itk == 5.4.*` and bump the package to a new major version.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,109 @@
+[build-system]
+requires = ["scikit-build-core"]
+build-backend = "scikit_build_core.build"
+
+[project]
+name = "itk-iomeshswc"
+version = "1.0.0"
```
Comment (thewtex) [inline pyproject.toml]:
```suggestion
version = "0.3.0"
```

Comment (jhlegarreta) [reply pyproject.toml]: Done in ce9565d.

**Comment by hjmjohnson** [comment]
@jhlegarreta Is this effort abandoned?

**Comment by jhlegarreta** [comment]
> @jhlegarreta Is this effort abandoned?

It is not; adopting `pyproject.toml` for project configuration should work. Errors are related to the `ITKRemoteModuleBuildTestPackageAction` version:
https://github.com/InsightSoftwareConsortium/ITKIOMeshSWC/actions/runs/12020402905/job/33508940808#step:1:36
https://github.com/InsightSoftwareConsortium/ITKIOMeshSWC/actions/runs/12020402905/job/33508941162#step:8:171

**Comment by jhlegarreta** [comment]
So to be clear, there is no further work required here: it should be safe to be merged once PR #5 gets merged and this one is rebased on `main`.

**Comment by jhlegarreta** [comment]
@hjmjohnson Checks are passing.

**Pull request merged**